guides/v2.1/coding-standards/docblock-standard-general.md

```diff
@@ -539,39 +546,29 @@ Documentation is organized with the use of namespaces.
 ### @inheritdoc tag
 {:#inheritdoc}
 
-Whenever possible the `@inheritdoc` tag MUST be used for children to avoid duplication of doc blocks.
-
-Even Though PHPDocumentor understands inheritance and uses the parent doc block by default (without `@inheritdoc` tag specified), including the tag helps ensure that the doc block is not missed at all.
-
-Rules for usage of the tag:
-
-* Use `@inheritdoc` (notice no braces around) to indicate that the entire doc block should be inherited from the parent method.
-* Use the inline `{@inheritdoc}` tag (with braces around) in long descriptions to reuse the parent's long description. The tagged method MUST have its own short description.
+The `@inheritdoc` tag SHOULD NOT be used.
+If a child class method requires a long description to explain its purpose, it may use `@inheritdoc` to indicate the new description is intended as an addition to the parent method description.
+In general such method overrides are a code smell and should be used as an incentive to make the code more self-documenting if possible.
 
 **DocBlock for the Interface**
+
 ```php
 /**
  * Interface for mutable value object for integer values
  */
 interface MutableInterface
 {
     /**
-     * Get value
-     *
      * Returns 0, if no value is available
-     *
-     * @return int
      */
-    public function getVal();
+    public function getVal(): int;
 
     /**
-     * Set value
-     *
      * Sets 0 in case a non-integer value is passed
-     *
-     * @param int $value
+     * 
+     * @param mixed $value
```

Comment:
I believe `mixed` must be avoided in favor of specific types (even if there are 3-4 of them).

Reply:
But what if they are an infinite number of possible types, for example anything that can be cast to a string?

Reply:
True :) "should be avoided whenever possible" then. Still, I would get rid all of `mixed` from code samples in doc though.

Reply:
@Vinai,

> But what if they are an infinite number of possible types

Changed my mind. Some interface or pseudo-type must be used. If there are too many different classes possible, just use `object`.

So, I would forbid `mixed` completely but this and few other my items can be added to separate PR so that this one only concentrates on stripping redundant declarations.

Reply:
I got rid of `mixed` annotations in the examples.